samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -82,17 +82,19 @@
    * non-stream systems (e.g., an external database).
    *
    * @param sinkFn the function to send messages in this stream to an external system
+   * @return this {@link MessageStream} object
    */
-  void sink(SinkFunction<M> sinkFn);
+  MessageStream<M> sink(SinkFunction<? super M> sinkFn);
```

Comment:
Issue: I think we decided to make these terminal because sink and sendTo can repartition the output stream while this returns the un-repartitioned stream.